pkg/utils/kubelessutil.go

```diff
@@ -684,6 +684,25 @@ func EnsureFuncDeployment(client kubernetes.Interface, funcObj *kubelessApi.Func
 		}
 	}
 
+	// Add soft pod anti affinity
+	dpm.Spec.Template.Spec.Affinity = &v1.Affinity{
+		PodAntiAffinity: &v1.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{
+				{
+					Weight: 100,
+					PodAffinityTerm: v1.PodAffinityTerm{
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"function": funcObj.ObjectMeta.Name,
```

Comment:
I would also add the label: `created-by: kubeless` to be sure

Reply:
Makes sense, added.